### PR TITLE
[CARBONDATA-1751] Modify sys.err to AnalysisException when uses run related operation except IUD,compaction and alter

### PR DESCRIPTION
**Pull request by xubo245**
carbon printout improper error message, for example, it printout system error when users run create table with the same column name, but it should printout related exception information

So we modify sys.error method to AnalysisException when uses run related operation except IUD,compaction and alter

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
     NO
 - [x] Any backward compatibility impacted?
     No
 - [x] Document update required?
    No
 - [x] Testing done
        CarbonTableSchemaCommonSuite
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
     MR68

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1203/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1209/

**Comment by xubo245** [comment]
Please review it @jackylk  @QiangCai

File: integration/spark-common-test/src/test/scala/org/apache/spark/sql/execution/command/CarbonTableSchemaCommonSuite.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        s"""
+           | CREATE TABLE carbon_table(
+           | BB INT, bb char(10)
+           | )
+           | STORED BY 'carbondata'
+       """.stripMargin)
+    } catch {
+      case ex: AnalysisException => Assert.assertTrue(true)
+      case ex: Exception => Assert.assertTrue(false)
+    }
```
Comment (jackylk) [inline integration/spark-common-test/src/test/scala/org/apache/spark/sql/execution/command/CarbonTableSchemaCommonSuite.scala]: If no exception, testcase should fail

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1224/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1239/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1254/

**Comment by xubo245** [comment]
Please review it @jackylk

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala [hunk 2]
```diff
@@ -662,16 +663,18 @@ object CommonUtil {
...
       if (columnCountInSchema >= maxColumnsInt) {
-        sys.error(s"csv headers should be less than the max columns: $maxColumnsInt")
+        CarbonException.analysisException(
+          s"csv headers should be less than the max columns: $maxColumnsInt")
       } else if (maxColumnsInt > CSVInputFormat.THRESHOLD_MAX_NUMBER_OF_COLUMNS_FOR_PARSING) {
-        sys.error(s"max columns cannot be greater than the threshold value: ${
-          CSVInputFormat.THRESHOLD_MAX_NUMBER_OF_COLUMNS_FOR_PARSING
-        }")
+        CarbonException.analysisException(
+          s"max columns cannot be greater than the threshold value: ${
```
Comment (jackylk) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala]: move 
```
${
 +            CSVInputFormat.THRESHOLD_MAX_NUMBER_OF_COLUMNS_FOR_PARSING
 +          }"
```
into one line, like line 667. the same for line 677

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1259/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1263/

File: integration/spark-common-test/src/test/scala/org/apache/spark/sql/execution/command/CarbonTableSchemaCommonSuite.scala (hunk 1 above)
Comment (xubo245) [reply integration/spark-common-test/src/test/scala/org/apache/spark/sql/execution/command/CarbonTableSchemaCommonSuite.scala]: Ok, I have fixed it.

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala (hunk 2 above)
Comment (xubo245) [reply integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala]: Ok, I have modified it.

**Pull request closed without merging**